server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -0,0 +1,97 @@
+package services.ti;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import forms.UpdateApplicantDob;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import models.Account;
+import models.Applicant;
+import models.TrustedIntermediaryGroup;
+import play.data.Form;
+import repository.SearchParameters;
+import repository.UserRepository;
+import services.DateConverter;
+import services.applicant.exception.ApplicantNotFoundException;
+
+public class TrustedIntermediaryService {
+
+  private final DateConverter dateConverter;
+  private final UserRepository userRepository;
+
+  @Inject
+  public TrustedIntermediaryService(DateConverter dateConverter, UserRepository userRepository) {
+    this.dateConverter = Preconditions.checkNotNull(dateConverter);
+    this.userRepository = Preconditions.checkNotNull(userRepository);
+  }
+
+  public ImmutableList<Account> retriveSearchResult(
+      SearchParameters searchParameters,
+      Optional<TrustedIntermediaryGroup> trustedIntermediaryGroup) {
+    return filterAccountsBySearchParams(searchParameters, trustedIntermediaryGroup.get());
+  }
+
+  public ImmutableList<Account> filterAccountsBySearchParams(
+      SearchParameters searchParameters, TrustedIntermediaryGroup tiGroup) {
+    ImmutableList<Account> allAccounts = tiGroup.getManagedAccounts();
+    allAccounts =
+        allAccounts.stream()
+            .filter(
+                account -> {
```

Comment:
Based on the method signatures, I'd expect this to return an account if it matches by name OR by date of birth. It appears that presently it will early out if a name parameter is provided that doesn't match the applicant's name.

I think this can be fixed fairly quickly by only returning true if it's a match.

Additionally, this could benefit from a specific regression test.